### DISCUSSION TITLE
Linked Issues/PRs added to a Projects (beta) board should show as a single linked entry

### DISCUSSION THREAD

**Discussion by 0xdevalias** [community/community#4714] [Projects and Issues] [answered]
Currently we can add both an issue, and a PR to the project (beta) board as separate entries, but this ends up:

- cluttering up the board by having 2+ entries for the 'same thing', or
- not allowing us to link one of the items to the board/track it's status/etc

On the old project boards (from memory) there was a 'collapsible' view that would show all PRs linked to an issue. This was ok, but still had the side effect of the PR itself not showing a 'reference' to the project board. So to get that '2-way-link', we would always end up adding the PR to the board individually as well, thus ending up in the same 'duplicated items on the board' situation as described above.

**Comment by zhyatt** [comment]
Came here to see if others faced this same issue, or if there were good workarounds or approaches to having both the Issue and PR added to a project, but not seeing the duplicate items. I am specifically using the non-beta Project boards, but hoping to get access to beta soon. Commenting and upvoting in hopes this gets more attention.

**Comment by kevinwhoffman** [comment]
> Currently we can add both an issue, and a PR to the project (beta) board as separate entries, but this ends up:
>
> - cluttering up the board by having 2+ entries for the 'same thing', or
> - not allowing us to link one of the items to the board/track it's status/etc

I'm facing this exact challenge and I've also considered adding both the issue and the pull request to the project, however this is not ideal as it clutters the board and inevitably we miss pull requests when developers forget to add them to the project.

Also note there is related feedback here:
- https://github.com/github/feedback/discussions/5003

**Comment by jorenbroekema** [comment]
Facing the same issue. I'm considering doing some automation step so that if a PR references an issue with "closes"/"fixes", I will check my projects board to see if PR / issue are there as duplicates, if so, remove the issue and keep the PR. 

The only problem with this is that having a PR in the board that references an issue that it will close, it does not visually show you that there's one or multiple linked issues that the PR addresses, so then it may seem as if the issue that was "deduped" was removed and is now gone altogether, but it's actually just hidden.

So basically I'd love having
- this deduplication to be done automatically, perhaps allow the user to decide whether it should keep the PR or keep the issue. 
- PRs in boards to visually show linked issues and allow me to navigate to them with a single click from the board.

**Comment by michael-mwp** [comment]
For my team, the legacy project board approach of listing linked PRs on each issue card worked really well. The board is uncluttered and we can quickly find and see the state of linked PRs.

The new beta project board doesn't show PRs linked to issues. During stand-up we have to navigate to an issue if we want to check and discuss linked PRs, which isn't as good an experience.

Explicitly linking PRs to a project isn't necessary as the PR links to an issue which states the project, so the PR is implicitly associated with the project. But, it could be useful if project automation explicitly linked the PR to the issue's project. I can't see a need for a PR and linked issue to be in separate projects, so automation could work fine.

We'd really like to see the current project approach of embedding the linked PRs in a project issue card replicated in projects beta.

**Comment by azenMatt** [comment, marked as answer]
Hi everyone - we just released this feature! 

You can check it out here - https://github.blog/changelog/2021-12-16-the-new-github-issues-12-16-update/

![changelog_-_linked_prs_v2 2c](https://user-images.githubusercontent.com/7584089/146577210-abb56ee5-37da-45eb-81a6-0b05f92fbb48.gif)

**Reply by zhyatt** [reply]
@azenMatt In testing this feature on a new board, I can see the linked PR on the Issue, but when both the PR and Issue are added to the board, they have separate entries instead of being collapsed into a single entry (the Issue only showing with the related PR tag included). So although this update seems to bring parity with the old style boards for showing PRs related to Issues, it doesn't seem to solve the original issue raised, which is looking for the removal of duplicate related entries.

Am I correct in that?

**Reply by dclear** [reply]
@azenMatt thanks for this; I am trying to make this work with [with keyword linking](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), where the PR is one repo and the issue is in another. Is that supported?

**Reply by jayspadie** [reply]
@zhyatt, you are correct that this feature to show linked pull requests for tracked issues does not prevent users from redundantly adding pull requests to projects as well, and it will not collapse the two entries into one since field values could still vary between an issue and linked pull requests.

Personally, I would advocate for adding only the issue to the project and linking the issue and related pull request(s).  This avoids duplication in the project, though to navigate from the pull request UI to any related projects, one would need to click through the linked issue(s) first.  Navigating from the project to any linked pull requests is exposed via the new Linked Pull Request column for the tracked issues.

Of course, we leave it to users to define their own processes and practices.  If they would prefer to add both issues and pull requests to projects so that they have a direct link from the pull request UI to the project (as well as additional field values specific to those pull requests), then they can reduce the duplication in the project views using filters such as “is:issue” or “is:pr”.  In that case, users could have a view that shows only issues and their linked pull requests, and/or they could have a view that shows only pull requests associated with the project.

@dclear, yes, if an issue is linked to a pull request via keyword linking, even across repositories, then that pull request should be visible in the Linked Pull Request column for that issue.

**Reply by zhyatt** [reply]
@jayspadie Thanks. Based on your response, I assume the original request to prevent both issue and related PR showing on a board they are both added to is not viable and not planned as a feature to be added. I think the confusion came from @azenMatt saying this feature was just released and marking this as the answer. I appreciate all the efforts in making these tools the best they can be!

**Reply by jayspadie** [reply]
Thanks, @zhyatt.  Yeah, I believe @azenMatt simply replied to a few discussions relating to the gap we had with beta Projects missing linked pull request information, and for many users this is exactly what they needed.  His reply was not specifically to this feature request to determine that an issue and pull request are linked and collapse them to a single entry.  I'm not aware of any near-term plans to do that.

Hopefully the optional use of persisted, filtered views for issues or pull requests will work to remove duplication if you choose to continue to add them both.  Or more generally, you could create another custom field with a Hide option and filter out any item on that.  Anyway, thanks again for your feedback and please continue letting us know how things could be improved.

**Comment by mxa0079** [comment]
I think this behavior is very confusing for newcomers. My team has struggled with this and so I came looking for a solution.

Id love if the project had the ability to collapse a PR and and Issue into a single item in the board.

Or maybe I am not looking at the value of this. What is the logic behind keeping them separate? Is something that was thought through, or is it one of those "it is just the way it is" things? In which scenario would it be useful to have an issue and the PR linked to that issue as separate cards in the board?

**Reply by jayspadie** [reply]
Keep in mind that you can filter any view to exclude PRs by including `-is:pr` in the filter bar.  Then you'll have a board view that includes only issues (and draft items) with any linked PRs showing only as a pill within the issue card.  Does that help?

**Reply by mxa0079** [reply]
It helps. but I do not think it address the root issue. Even after filtering, you have to keep moving across your board two items (the PR and the issue), or you have to make sure that no PRs are added to a project. To me it seems like a design flaw, but I may be missing something.

**Reply by jayspadie** [reply]
> Even after filtering, you have to keep moving across your board two items (the PR and the issue)

I'm a little confused by this one.  If you include `-is:pr` on your board filter, then you'll no longer see pull requests to move.  You'll only have the issues on the board.  Can you clarify please?

If you do the above, then if you want, you can create another table view that shows only PRs with `is:pr`, sorted by Status.  If you also set your `item added to project` and `pull request merged` workflows to set the Status field, then you're left with a view that shows newly added PRs (in progress or ready for review) at the top.  No other manual moves are necessary if you don't care to track the status of PRs with more granularity than that.

I'm not saying it's perfect, but it was built to be flexible.  Some teams choose to encourage adding PRs to projects for the additional visibility and traceability, while others only add issues.  In either case, you can update view filters to show only the type that you want to see.

**Reply by kevinwhoffman** [reply]
My workflow improved a lot when I excluded PRs from the project, or at least filtered them out using `-is:pr` as @jayspadie suggested. The issue gets moved from one stage to the next (Open > In Progress > Review > Closed). Any linked PRs go along for the ride as they appear on the issue card or row.
